src/main/java/de/picturesafe/search/elasticsearch/impl/StaticIndexPresetConfigurationProvider.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright 2020 picturesafe media/data/bank GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package de.picturesafe.search.elasticsearch.impl;
+
+import de.picturesafe.search.elasticsearch.IndexPresetConfigurationProvider;
+import de.picturesafe.search.elasticsearch.config.IndexPresetConfiguration;
+
+import java.util.List;
+import java.util.Map;
+import java.util.TreeMap;
+
+public class StaticIndexPresetConfigurationProvider implements IndexPresetConfigurationProvider {
+
+    private final Map<String, IndexPresetConfiguration> indexPresetConfigurationByAlias = new TreeMap<>();
+
+    public StaticIndexPresetConfigurationProvider(List<IndexPresetConfiguration> indexPresetConfigurations) {
+        indexPresetConfigurations.forEach(conf -> indexPresetConfigurationByAlias.put(conf.getIndexAlias(), conf));
+    }
+
+    @Override
+    public IndexPresetConfiguration getIndexPresetConfiguration(String indexAlias) {
+        throw new RuntimeException("Operation getIndexPresetConfiguration not supported!"); // TODO!
```

Comment:
Shouldn't this:
return indexPresetConfigurationByAlias.get(indexAlias);
?

Reply:
Oops!